misc/mp_assert.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * This file is part of mpv.
+ *
+ * mpv is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/**
+ * @brief This file provides two macros, mp_assert and mp_require, used to check
+ * for conditions that should never occur.
+ *
+ * The mp_assert macro is a convenience wrapper around the standard assert,
+ * which also sinks unused variables to prevent compiler warnings. It should be
```

Comment:
so any side effects will still happen, not sure if that's a good idea

Reply:
I don't think so. It's using `sizeof(expr)` to silence unused variables. `sizeof` doesn't evaluate anything. (Unless it's vla, but mpv builds with `-Wvla` already so that's mostly covered for).

Reply:
That's correct. The `sizeof` trick suppress compiler warnings, but nothing is in fact evaluated during runtime.